src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileInfo.cs

```diff
@@ -34,7 +29,9 @@ public PhysicalFileInfo(FileInfo info)
 
         public Stream CreateReadStream()
         {
-            var bufferSize = (int)Math.Min(MaxBufferSize, Math.Max(MinBufferSize, Length));
+            // We are setting buffer size to 1 to prevent FileStream from allocating it's internal buffer
+            // 0 causes constructor to throw
+            var bufferSize = 1;
```

Comment:
This could be an argument to to physical file system provider. That way if somebody wants more control they can get that via this argument.